Fix bat scripts to work with JAVA_HOME containing special characters

pgomulka opened the pull request:
the `elasticsearch.bat` and `elasticsearch-env.bat` won't work if `JAVA` contains parentheses. This seems to be the limitation of `FOR /F IN (command) DO` syntax.  
The `JAVA` variable present in a command contains a path to a binary to start elasticsearch (with spaces & parans). We can workaround the problem of spaces and parentheses in this path by referring this variable with a `CALL` command.
Note that executing binaries with `CALL` is an undocumented behaviour (but works)

closes #38578
closes https://github.com/elastic/elasticsearch/pull/38624
closes https://github.com/elastic/elasticsearch/pull/33405
closes https://github.com/elastic/elasticsearch/issues/30606

elasticmachine commented:
Pinging @elastic/es-core-infra

pgomulka commented:
ok to test

csetzkorn commented:
This:

bin\elasticsearch-plugin install file:///C:/bla/ingest-attachment-6.6.0.zip

still throws an error:

\Common was unexpected at this time.

pgomulka commented:
@csetzkorn thank you for this - will indeed increase the scope of this PR to include fixes to all windows bat scripts using to JAVA_HOME with '('

pgomulka commented:
@csetzkorn can you please confirm that you had the failure you mentioned after applying the patch from this PR?

pgomulka commented:
@elasticmachine run elasticsearch-ci/packaging

pgomulka committed:
Bat scripts to work with JAVA_HOME with parantheses

the elasticsearch.bat and elasticsearch-env.bat won't work if JAVA
contains parentheses. This seems to be the limitation of FOR /F IN
(command) DO syntax.
The JAVA variable present in a command contains a path to a binary to
start elasticsearch (with spaces & parans). We can workaround the
problem of spaces and parentheses in this path by referring this
variable with a CALL command.
Note that executing binaries with CALL is an undocumented behaviour (but
works)

pgomulka commented:
closing as cannot resolve the conflicts after the base change (seems to be a github bug)
